lib/dx/evaluation.ex

```diff
@@ -23,10 +23,12 @@ defmodule Dx.Evaluation do
   end
 
   def from_options(opts) do
+    {opts, loader_opts} = Keyword.pop(opts, :loader, [])
+
     %__MODULE__{}
     |> add_options(opts)
     |> case do
-      %{cache: nil} = eval -> Map.put(eval, :cache, eval.loader.init())
+      %{cache: nil} = eval -> Map.put(eval, :cache, eval.loader.init(loader_opts))
```

Comment:
I'm thinking whether it might be good to store the options, so it's easier to access them again, like for error messages (the timeout; see my https://github.com/dx-beam/dx/pull/20#discussion_r1168433741). I'm not sure.

Then this could be either

```suggestion
      %{cache: nil, loader_options: opts} = eval -> Map.put(eval, :cache, eval.loader.init(opts))
```

or

```suggestion
      %{cache: nil} = eval ->
        opts = Map.take(eval, [:timeout, :repo_options]) |> Enum.to_list()
        Map.put(eval, :cache, eval.loader.init(opts))
```

depending on the options structure we choose (see above).